go/vt/wrangler/reparent.go

```diff
@@ -539,9 +539,9 @@ func (wr *Wrangler) chooseNewMaster(
 	avoidMasterTabletAlias *topodatapb.TabletAlias,
 	waitSlaveTimeout time.Duration) (*topodatapb.TabletAlias, error) {
 
-	if avoidMasterTabletAlias == nil {
-		return nil, fmt.Errorf("tablet to avoid for reparent is not provided, cannot choose new master")
-	}
+	// if avoidMasterTabletAlias == nil {
```

Comment:
If this is nil, we need to set `avoidMasterTabletAlias = shardInfo.MasterAlias` , as `chooseNewMaster` requires that we have something passed in as that field, yes?

Reply:
I think I agree with @zmagg. The simplest way would be to leave the existing logic in `plannedReparentShardLocket` as is, and perform this assignment before all the checks.

And then we can delete this check here.